.github/workflows/e2e-upgrade.yaml

```diff
@@ -131,3 +131,10 @@ jobs:
       relayer-type: rly
       relayer-image: ghcr.io/cosmos/relayer
       relayer-tag: latest
+
+  slack-notify:
+    needs: [upgrade-v5-hermes, upgrade-v7-hermes, upgrade-v7_1-hermes, upgrade-v8-hermes, upgrade-v5-rly, upgrade-v7-rly, upgrade-v7_1-rly, upgrade-v8-rly]
+    uses: ./.github/workflows/slack-notify.yaml
+    secrets: inherit
+    # Only send a slack notification if any of the tests failed, run only on cron jobs
+    if: ${{ always() && contains(needs.*.result, 'failure') && github.event_name == 'schedule' }}
```

Comment:
This works fine but there is a built in `{{ failure() }}` [example](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-status-check-functions).

Should we use that instead?

Also, I don't think this lets us know which one failed (I don't think that's a huge problem since we will just click on the link and see for ourselves)

Reply:
> This works fine but there is a built in {{ failure() }} [example](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-status-check-functions).

does this also apply to jobs? I remember this is an annoying discrepancy and using the contains with `needs` was the only workaround to getting the behaviour "fail if any of these jobs failed".

> I don't think this lets us know which one failed

Yup, tried to do that first go around. Requires additional inputs to re-usable workflow and explicit passing of secrets with `inherit`, I didn't particularly like the way it was shaping out so went with the aggregate.

Reply:
Ah I missed that it was for a step and not a job! I'm not sure if it can be used the same way. Happy to leave it for now, either way it achieves the same result.